src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameCollector.cs

```diff
@@ -94,53 +122,178 @@ internal BlameCollector(IBlameReaderWriter blameReaderWriter, IProcessDumpUtilit
+                // if/when we try to kill the existing proc dump process.
+                this.processDumpUtility.DetachFromTargetProcess(this.testHostProcessId);
+            }
+
+            try
+            {
+                this.processDumpUtility.StartHangBasedProcessDump(this.testHostProcessId, this.attachmentGuid, this.GetResultsDirectory(), this.processFullDumpEnabled);
+            }
+            catch (Exception ex)
+            {
+                EqtTrace.Error($"BlameCollector.CollectDumpAndAbortTesthost: Failed with error {ex}");
+            }
+
+            try
+            {
+                var dumpFile = this.processDumpUtility.GetDumpFile();
+                if (!string.IsNullOrEmpty(dumpFile))
                 {
-                    if (string.Equals(attribute.Value, Constants.TrueConfigurationValue, StringComparison.OrdinalIgnoreCase) || string.Equals(attribute.Value, Constants.FalseConfigurationValue, StringComparison.OrdinalIgnoreCase))
-                    {
-                        bool.TryParse(attribute.Value, out this.collectDumpAlways);
-                    }
-                    else
-                    {
-                        this.logger.LogWarning(this.context.SessionDataCollectionContext, string.Format(CultureInfo.CurrentUICulture, Resources.Resources.BlameParameterValueIncorrect, attribute.Name, Constants.TrueConfigurationValue, Constants.FalseConfigurationValue));
-                    }
+                    var fileTransferInformation = new FileTransferInformation(this.context.SessionDataCollectionContext, dumpFile, true, this.fileHelper);
+                    this.dataCollectionSink.SendFileAsync(fileTransferInformation);
                 }
-                else if (string.Equals(attribute.Name, Constants.DumpTypeKey, StringComparison.OrdinalIgnoreCase))
+                else
                 {
-                    if (string.Equals(attribute.Value, Constants.FullConfigurationValue, StringComparison.OrdinalIgnoreCase) || string.Equals(attribute.Value, Constants.MiniConfigurationValue, StringComparison.OrdinalIgnoreCase))
-                    {
-                        this.processFullDumpEnabled = string.Equals(attribute.Value, Constants.FullConfigurationValue, StringComparison.OrdinalIgnoreCase);
-                    }
-                    else
-                    {
-                        this.logger.LogWarning(this.context.SessionDataCollectionContext, string.Format(CultureInfo.CurrentUICulture, Resources.Resources.BlameParameterValueIncorrect, attribute.Name, Constants.FullConfigurationValue, Constants.MiniConfigurationValue));
-                    }
+                    EqtTrace.Error("BlameCollector.CollectDumpAndAbortTesthost: blame:CollectDumpOnHang was enabled but dump file was not generated.");
                 }
-                else
+            }
+            catch (FileNotFoundException ex)
+            {
+                EqtTrace.Error(ex);
```

Comment:
More descriptive message here.

Reply:
ideally it should always be there, this is not expected to happen. hence throwing default. copied code below btw :p